File: ai/all_heuristics.rb
# When creating a new heuristic, remember to add it here.

require_relative "spacer"
require_relative "executioner"
require_relative "savior"
require_relative "hunter"
require_relative "connector"
require_relative "pusher"
require_relative "no_easy_prisoner"

class Heuristic
  def Heuristic.all_heuristics
    return [Spacer, Executioner, Savior, Hunter, Connector, Pusher, NoEasyPrisoner]
  end
end

File: js/ai/AllHeuristics.js
//Translated from all_heuristics.rb using babyruby2js
'use strict';

// When creating a new heuristic, remember to add it here.
var Spacer = require('./Spacer');
var Savior = require('./Savior');
var Hunter = require('./Hunter');
var Connector = require('./Connector');
var Pusher = require('./Pusher');
var NoEasyPrisoner = require('./NoEasyPrisoner');
var Shaper = require('./Shaper');


var allHeuristics = function () {
    return [
        NoEasyPrisoner,
        Savior,
        Hunter,
        Connector,
        Spacer,
        Pusher,
        Shaper
    ];
};
module.exports = allHeuristics;
